### ISSUE TITLE
[ENHANCEMENT] More descriptive default file name when saving a json conversation

### ISSUE THREAD

**Issue by NintendoManiac64** [kaihordewebui/kaihordewebui.github.io#12]
Currently the default file name is just `saved_story.json` which, let's be honest, isn't exactly all that descriptive.

A simple way to make it more descriptive is to have it begin with a timestamp in ISO8601 format (year-month-day hours꞉minutes:seconds.milliseconds) followed by the scenario chosen (or something like "custom scenario" if not using one of the stock ones).

So in total it would look something like:

- 2023-04-13 19꞉27.123 Haruka.json
- 2023-04-13T19꞉27.123 Fantasy Isekai.json
- 2023-04-13 19꞉27.123　Fantasy Isekai - saved-story.json

(note that, since Windows doesn't like having colons for file names, I've replaced all instances of colons with the "[Modifier Letter Colon (U+A789)](https://unicodeplus.com/U+A789)" which looks nearly identical and is compatible with WIndows' file names)

**Comment by LostRuins** [comment]
Actually that's not the case, it's only `saved_story.json` if you started a new story without loading an existing one. Otherwise, it will attempt to save back to the previous filename. Unfortunately, browser limitations means that the UI cannot determine the actual name of the `.json` when it saves, only when it loads.

Autogenerating a name may cause issues for people who want to overwrite their existing story though. I'd imagine many people not wanting to deal with multiple files and versions, but just continuing their progress on the same file. By generating a new timestamp each time, you always get a new filename each time you save. 

I think it's probably best to keep the name consistent, and allow the OS to deduplicate if the file already exists.

**Comment by NintendoManiac64** [comment]
> Actually that's not the case, it's only `saved_story.json` if you started a new story without loading an existing one.

I am aware of this - that is why I said *default* file name.

Maybe I should have been more clear in that I am not at all talking about when you've previously loaded a .json file.

**Comment by henk717** [comment]
Couple possible idea's.
- Story / instruct / adventure mode, probably best to leave as is or pull an MS word and make it the first sentence that isn't the instruction tag.
- Chat mode, character name.

**Comment by fuhrriel** [comment]
Would it be possible to add a text field to the save/load box that would be the name for the downloaded file? I'd rather download it already named the correct way, rather than having to open my trash of a Downloads directory and find the file there.

**Comment by LostRuins** [comment]
Yes, you can do that. Enable Rename save file
![image](https://github.com/user-attachments/assets/adef8735-7630-437e-ab58-9600d2006095)
![image](https://github.com/user-attachments/assets/0155d6f4-76ff-4b67-b78f-dea2d8e8f5a5)

**Comment by fuhrriel** [comment]
Yeah, that solves my issue, but I don't think that's the best place for this option to be. Way too hidden.
